api/tests/Api/ECampApiTestCase.php

```diff
@@ -74,6 +74,10 @@ public static function deepCloneArray(array $array): mixed {
         return json_decode($json_encode, true);
     }
 
+    protected static function getKernelClass(): string {
+        return TestKernel::class;
+    }
```

Comment:
If you really need to do this, you could instead create a services_test.yaml file next to services.yaml. No need for a custom kernel class calling a compiler pass just to make some simple adjustments to services.yaml.
https://symfony.com/doc/current/testing.html

That will probably affect all tests, also unit tests, but services.yaml is primarily used in integration tests, so it shouldn't disrupt too much.

Reply:
That was first idea too, the question is: which service?
when i overrode the existing service, it did not merge the definitions, but set all the other definitions to the default.
when i made our decorators public, this did not work too.

It extends the kernel from production, and just makes some services available in the tests (which would have been handy in other tests too).
I could use this kernel only in this test, if you think that is less intrusive.

Reply:
See above, i did this for api-platform 3.2

And:
"This means that if you have a private service not used by any other service, Symfony will remove it and you won't be able to get it as explained in this article. "